### DISCUSSION TITLE
Is there a way to include danger JS files from other danger JS files?

### DISCUSSION THREAD

**Discussion by jaymzh** [danger/danger-js#1417] [Q&A] [answered]
The ruby version has an `import` function that will include other Danger files - does the JS have an equivalent?

**Comment by orta** [comment]
`require` or `import` should work

**Reply by jaymzh** [reply]
@orta - nope...

```
Unable to evaluate the Dangerfile
 dangerfile.js:1[6](https://github.com/xxx/xxx/actions/runs/7185740953/job/19569640406?pr=58731#step:4:7)
require './api/dangerfile.js'
        ^^^^^^^^^^^^^^^^^^^^^
```

```
Cannot use import statement outside a module
dangerfile.js:16
import './api/dangerfile.js'
^^^^^^
```

**Reply by jaymzh** [reply]
Hmm `require('...')` gets me further, but then it complains about my `import { danger } from 'danger'`... this probably is stuff a normal JS person would just know, I guess, but I'm not super familiar with.

**Reply by fbartho** [reply, marked as answer]
Only the "parent" Dangerfile can directly `import { danger } from 'danger'`

This is briefly documented here: https://github.com/danger/danger-js/blob/main/docs/usage/extending-danger.html.md#writing-your-plugin

Specifically:
>Note, that you should not add a import { danger, et, cetera } from "danger" in your plugin. The dirty truth is that Danger completely ignores that at runtime, and just puts all of its exports inside the global context. You'll get compiler errors, or confusing shadow variable issues. So assume that it's there, and working fine

If you're using JS, and not Typescript, then you can just remove the "import…from 'danger'" lines, and it should work.

If you're using TypeScript, you may need some lies for the compiler to know that `danger` is globally available.

**Reply by jaymzh** [reply]
awesome, thanks!
